tests/src/lib.rs

```diff
@@ -471,62 +492,67 @@ mod utils {
         _namespace: Option<ResourceGuard>,
     }
 
-    /// randomize_name: should a random suffix be added to the end of resource names? e.g.
-    ///                 for `echo-service`, should we create as `echo-service-ybtdb`.
-    /// delete_after_fail: delete resources even if the test fails.
+    /// Create a new [`KubeService`] and related Kubernetes resources. The resources will be deleted
+    /// when the returned service is dropped, unless it is dropped during panic.
+    /// This behavior can be changed, see [`FORCE_CLEANUP_ENV_NAME`].
+    /// * `randomize_name` - whether a random suffix should be added to the end of the resource
+    ///   names
     #[fixture]
     pub async fn service(
         #[default("default")] namespace: &str,
         #[default("NodePort")] service_type: &str,
         #[default("ghcr.io/metalbear-co/mirrord-pytest:latest")] image: &str,
         #[default("http-echo")] service_name: &str,
         #[default(true)] randomize_name: bool,
-        #[default(false)] delete_after_fail: bool,
```

Comment:
I would leave this there and then OR it with the env var. Some deployments do work as long as they exist, so I would rather not have them keep running on fail by default.

Alternatively, we could make deleting everything the default, and only keep the resources when an env var is set. If you want to have it that way, please consult the team before making the change.

Reply:
The team seems to like the second idea, I changed the flow